x-pack/legacy/plugins/apm/server/lib/transactions/breakdown/index.ts

```diff
@@ -174,12 +174,38 @@ export async function getTransactionBreakdown({
           ...p,
           [name]: p[name].concat({
             x: time,
-            y: percentage
+            y: percentage,
+            hasData: percentage !== null
           })
         };
       }, prev);
+
+      const lastValues = Object.values(updatedSeries).map(last);
+
+      // If for a given timestamp, some series have data, but others do not,
+      // we have to set any null values to 0 to make sure the stacked area chart
+      // is drawn correctly.
+      // If we set all values to 0, the chart always displays null values as 0,
+      // and the chart looks weird.
+      const hasAnyValues = lastValues.some(value => value.y !== null);
+      const hasNullValues = lastValues.some(value => value.y === null);
+
+      if (hasAnyValues && hasNullValues) {
+        Object.values(updatedSeries).forEach(series => {
+          const value = series[series.length - 1];
+          const isEmpty = value.y === null;
+          if (isEmpty) {
+            value.y = 0;
+          }
+        });
+      }
```

Comment:
The fix would have to be in a different place, because the fix we already have only operates on one series. To fix the issue in this PR, we need to operate on all series (we transform values for timestamp n in all series). We _could_ fix it in this component though, in `getVisSeries`: https://github.com/elastic/kibana/blob/304f702764a02813bc0d9265b7909e241d2760a0/x-pack/legacy/plugins/apm/public/components/shared/charts/CustomPlot/StaticPlot.js#L33-L38

Would that be okay too? I do agree it's a better place to solve it; the fix is closer to the cause (which is in `react-vis`) and we don't have to change our endpoint once `react-vis` either fixes it or we replace it with another charting library.

Reply:
Yeah, I think that makes sense 👍

Reply:
discussed via zoom to just set any null values to 0 if other span types have data for that timestamp. it's a lot easier and there's no real difference to the user between 0 and N/A for this specific situation.